frontend/src/components/Radio/Radio.tsx

```diff
@@ -0,0 +1,152 @@
+/**
+ * This file is a slightly modified version of Chakra UI's internal Radio
+ * implementation, which can be found here:
+ * https://github.com/chakra-ui/chakra-ui/blob/main/packages/radio/src/radio.tsx
+ *
+ * Limitations of Chakra's Radio mean that we cannot implement our desired
+ * design with the existing implementation. In particular, the "disabled"
+ * attribute does not apply to the label which wraps the component, meaning
+ * we cannot apply the correct styles to the Radio container when the button
+ * inside it is disabled (e.g. { cursor: 'not-allowed', bg: 'none' }).
+ *
+ * Hence this code is adapted to apply the desired styles to the label which
+ * wraps the component.
+ *
+ * The relevant issue in the Chakra UI repo is here:
+ * https://github.com/chakra-ui/chakra-ui/issues/4295
+ */
+
+import {
+  chakra,
+  forwardRef,
+  HTMLChakraProps,
+  layoutPropNames,
+  omitThemingProps,
+  SystemProps,
+  SystemStyleObject,
+  ThemingProps,
+  useMultiStyleConfig,
+  useRadio,
+  useRadioGroupContext,
+  UseRadioProps,
+} from '@chakra-ui/react'
+import { callAll, split } from '@chakra-ui/utils'
+
+type Omitted = 'onChange' | 'defaultChecked' | 'checked'
+type BaseControlProps = Omit<HTMLChakraProps<'div'>, Omitted>
+
+export interface RadioProps
+  extends UseRadioProps,
+    ThemingProps<'Radio'>,
+    BaseControlProps {
+  /**
+   * The spacing between the checkbox and its label text
+   * @default 0.5rem
+   * @type SystemProps["marginLeft"]
+   */
+  spacing?: SystemProps['marginLeft']
+  /**
+   * If `true`, the radio will occupy the full width of its parent container
+   *
+   * @deprecated
+   * This component defaults to 100% width,
+   * please use the props `maxWidth` or `width` to configure
+   */
+  isFullWidth?: boolean
+}
+
+/**
+ * Radio component is used in forms when a user needs to select a single value from
+ * several options.
+ *
+ * @see Docs https://chakra-ui.com/radio
+ */
+export const Radio = forwardRef<RadioProps, 'input'>((props, ref) => {
+  const { onChange: onChangeProp, value: valueProp } = props
+
+  const group = useRadioGroupContext()
+  const styles = useMultiStyleConfig('Radio', { ...group, ...props })
```

Comment:
use theme key here? but i think we can just do away with the theme key and use these strings instead :P

Reply:
this is actually a copypasta mistake haha oops. I actually like the theme keys because it's not immediately obvious to new devs that the string must be consistent with the one in the Chakra repo.